processor/src/operations/horner_ops.rs

```diff
@@ -0,0 +1,515 @@
+    /// the same instruction interleaved with any operation that loads the next batch of 8
+    /// coefficients on the top of the operand stack, i.e., `mem_stream` or `adv_pipe`.
+    ///
+    /// The stack transition of the instruction can be visualized as follows:
+    ///
+    /// Input:
+    ///
+    /// +------+------+------+------+------+------+------+------+---+---+---+---+------+------+------+------+
+    /// |  c0  |  c1  |  c2  |  c3  |  c4  |  c5  |  c6  |  c7  | - | - | - | - |c_addr|r_addr| acc1 | acc0 |
+    /// +------+------+------+------+------+------+------+------+---+---+---+---+------+------+------+------+
+    ///
+    ///
+    /// Output:
+    ///
+    /// +------+------+------+------+------+------+------+------+---+---+---+---+------+------+-------+-------+
+    /// |  c0  |  c1  |  c2  |  c3  |  c4  |  c5  |  c6  |  c7  | - | - | - | - |c_addr|r_addr| acc1' | acc0' |
+    /// +------+------+------+------+------+------+------+------+---+---+---+---+------+------+-------+-------+
+    ///
+    ///
+    /// Here:
+    ///
+    /// 1. ci for i in 0..=7 stands for the the value of the i-th coefficient in the current batch
+    ///    of 8 coefficients of the polynomial.
+    /// 2. (acc0, acc1) stands for an extension field element accumulating the values of the Horner
+    ///    evaluation procedure. (acc0', acc1') is the updated value of this accumulator.
+    /// 3. t_addr is the memory address from which we are loading the ci's when using the MSTREAM
+    ///    instruction or storing using the ADVPIPE instruction.
+    /// 4. r_addr is the memory address of the evaluation point i.e., alpha.
+    ///
+    /// The instruction also makes use of the helper registers to hold the value of
+    /// alpha = (alpha0, alpha1) during the course of its execution.
+    pub(super) fn op_horner_eval_base(&mut self) -> Result<(), ExecutionError> {
+        // --- read the values of the coefficients, over the base field, from the stack -----------
+        let coef = self.get_coeff_as_base_elements();
+
+        // --- read the randomness from memory ----------------------------------------------------
+        let alpha = self.get_random_point()?;
+
+        // --- read the accumulator values from stack ---------------------------------------------
+        let acc_old = self.get_accumulator();
+
+        // --- compute the updated accumulator values ---------------------------------------------
+        let acc_new =
+            coef.iter().rev().fold(acc_old, |acc, coef| QuadFelt::from(*coef) + alpha * acc);
+
+        // --- copy the stack ---------------------------------------------------------
+        self.stack.copy_state(0);
+
+        // --- update the accumulators ------------------------------------------------------------
+        self.stack.set(14, acc_new.to_base_elements()[1]);
+        self.stack.set(15, acc_new.to_base_elements()[0]);
```

Comment:
nit: can we have some constants at the top of the file for the indices used? It would make it easier to read so I don't have to refer back to the diagram to remember what `14` and `15` is.

e.g. `ACC_HIGH_INDEX` and `ACC_LOW_INDEX` or something like that

Reply:
added